### DISCUSSION TITLE
DV-Multiplayer v0.1.8.3 Beta

### DISCUSSION THREAD

**Discussion by AMacro** [AMacro/dv-multiplayer#38] [Announcements]
# Changes / Fixes
## UI
- Added an update alert when the game starts
- IPv4 lookup is now using another service to find external IPv4 addresses, old service proved unreliable for some players
- Fixed bug with the server not correctly extracting the IPv4 address when it was received from the IPv4 lookup service

## Game
- Jobs overviews now spawn, client can not take jobs yet.
- Fixed bug with MU cable connection sync - note MU acts glitchy on clients, but still works
- Fixed issue where new players/early game are forced to pay to rerail (vanilla game has $0 fee)

# Known Bugs
- Clients are unable to self service locomotives - self service objects (e.g. pipes, chutes, plugs, nozzles) not synced
- Horns, whistles and steam loco reversers can play up when more than one player is in the loco; this is not a breaking issue, only a mild inconvenience.
- MU chains behave the sane as the horns and whistles, playable but mildly annoying
- Shop, world items, inventory for players other than the host not yet synced
- Many other minor bugs and inconveniences, which we'll work through as we tick off the bigger issues.

# Features in progress
- Job sync - building framework for clients to take jobs. This depends on item sync
- Item sync - building framework to allow items to be sync'd between clients
- Get ping working in the server browser
- LAN discovery
- UPnP
- NAT punching

<hr /><em>This discussion was created from the release <a href='https://github.com/AMacro/dv-multiplayer/releases/tag/'>DV-Multiplayer v0.1.8.3 Beta</a>.</em>

**Comment by SeanJAnimations** [comment]
Hi! I've been testing with the mod with some friends, this version works quite well and it's been really fun! Do you have any potential ideas for when clients will be able to see job paperwork? And do you plan to change the character model anytime soon?

**Reply by AMacro** [reply]
Hey @SeanJAnimations,

We've been watching your videos when they come out; it's great seeing how much fun you guys are having!

In the current release, 0.1.8.3, clients should be able to see job orders, but not take them. There are some minor issues that may stop the orders from appearing sometimes - we were hoping to have job sync completed in this release, but we were made aware of some bugs that needed an immediate fix, so we released early.

The plan is 0.1.8.4 will have allbof job sync working, however, for it to work correctly we also need item syncing to be somewhat working.

As for timing, that is a little hard to predict, I will be away for most of October with limited opportunity to work on the mod. I hope we can have something released before then but can't make any promises.

Changing the character is low priority for us, we're not game devs or modellers, so our effort is better spent getting things working.
We're hoping someone in the community will create a model we can use, in which case it will be included much sooner, but failing that, once all the syncing is working we will look at making or buying a new character.

If you know anyone with modelling capabilities who'd be willing to donate a character, give them a nudge 😉. Bonus points if they are familiar with Unity Engine.

**Reply by SeanJAnimations** [reply]
Hey! Thanks for the reply! I'm glad you guys have your priorities straight and know what's important. And that all makes sense with the models. Thanks for the mod and we've been havin' a grand time, keep up the good work! :)

**Comment by SeanJAnimations** [comment]
Hello again! I just wanted to let ya know that this mod isn't compatible with "Passenger Jobs", on the client side. I'm not tryin' to bug ya with it like "gEt ThIs WoRkIn'", ya know. Just figured I should tell ya. Cars spawned by Passenger jobs will (for clients) appear to spawn on the same track for some reason, causing the cars to spaze out and it looks REAL weird, but the jobs and overviews seem to work fine. Again, thanks for the great mod! :)

**Reply by AMacro** [reply]
Yep, this one's a known issue, and we have some of the framework in place so it can be fixed in the future, but haven't looked too deeply into it yet.

It's one of the mods we will prioritise as it's very popular and we quite like it ourselves.

**Reply by SeanJAnimations** [reply]
Grand!

**Comment by Atheros524** [comment]
whenever loading into a friends carrer save this happens 
![Screenshot 2025-01-14 195545](https://github.com/user-attachments/assets/24d17df6-5948-4756-8b8e-88e9650848be)

**Reply by AMacro** [reply]
Could you please confirm you are playing B99 and using Multiplayer 0.1.9.7 (latest release)?

Please open an issue and upload your player.log (wait until you're stuck at 100% loaded, then copy the log file from `%appdata%\..\LocalLow\Altfuture\Derail Valley`)

**Comment by Atheros524** [comment]
b99 version 1.9.2, i wasn't aware there was a 0.1.9.7 version
